mods/quick-hide-desktop-icons.wh.cpp

```diff
@@ -0,0 +1,455 @@
+  $description:zh-CN: 如何触发切换 (鼠标/键盘/两者)
+  $description:zh-TW: 如何觸發切換 (滑鼠/鍵盤/兩者)
+  $description:ja-JP: トグルをトリガーする方法（マウス/キーボード/両方）
+  $options:
+  - mouse: Mouse only
+  - keyboard: Keyboard only
+  - both: Both
+  $options:zh-CN:
+  - mouse: 仅鼠标
+  - keyboard: 仅键盘
+  - both: 两者
+  $options:zh-TW:
+  - mouse: 僅滑鼠
+  - keyboard: 僅鍵盤
+  - both: 兩者
+  $options:ja-JP:
+  - mouse: マウスのみ
+  - keyboard: キーボードのみ
+  - both: 両方
+
+- mouseButton: left
+  $name: Mouse Button
+  $name:zh-CN: 鼠标按键
+  $name:zh-TW: 滑鼠按鍵
+  $name:ja-JP: マウスボタン
+  $description: Which mouse button to use
+  $description:zh-CN: 使用哪个鼠标按键
+  $description:zh-TW: 使用哪個滑鼠按鍵
+  $description:ja-JP: 使用するマウスボタン
+  $options:
+  - left: Left button
+  - right: Right button
+  - middle: Middle button
+  $options:zh-CN:
+  - left: 左键
+  - right: 右键
+  - middle: 中键
+  $options:zh-TW:
+  - left: 左鍵
+  - right: 右鍵
+  - middle: 中鍵
+  $options:ja-JP:
+  - left: 左ボタン
+  - right: 右ボタン
+  - middle: 中ボタン
+
+- clickType: double
+  $name: Click Type
+  $name:zh-CN: 点击类型
+  $name:zh-TW: 點擊類型
+  $name:ja-JP: クリックタイプ
+  $description: Single or double click
+  $description:zh-CN: 单击或双击
+  $description:zh-TW: 單擊或雙擊
+  $description:ja-JP: シングルクリックまたはダブルクリック
+  $options:
+  - single: Single click
+  - double: Double click
+  $options:zh-CN:
+  - single: 单击
+  - double: 双击
+  $options:zh-TW:
+  - single: 單擊
+  - double: 雙擊
+  $options:ja-JP:
+  - single: シングルクリック
+  - double: ダブルクリック
+
+- modifiers:
+  - ctrl: false
+  - alt: false
+  - shift: false
+  $name: Modifier Keys
+  $name:zh-CN: 修饰键
+  $name:zh-TW: 修飾鍵
+  $name:ja-JP: 修飾キー
+  $description: Require additional keys
+  $description:zh-CN: 需要额外按键
+  $description:zh-TW: 需要額外按鍵
+  $description:ja-JP: 追加のキーが必要
+- hotkey: F12
+  $name: Hotkey
+  $name:zh-CN: 热键
+  $name:zh-TW: 快捷鍵
+  $name:ja-JP: ホットキー
+  $description: Keyboard shortcut (when keyboard mode enabled)
+  $description:zh-CN: 键盘快捷键 (启用键盘模式时)
+  $description:zh-TW: 鍵盤快捷鍵 (啟用鍵盤模式時)
+  $description:ja-JP: キーボードショートカット（キーボードモード有効時）
+- enableSound: false
+  $name: Enable Sound
+  $name:zh-CN: 启用声音
+  $name:zh-TW: 啟用聲音
+  $name:ja-JP: サウンドを有効にする
+  $description: Play sound when toggling
+  $description:zh-CN: 切换时播放声音
+  $description:zh-TW: 切換時播放聲音
+  $description:ja-JP: トグル時にサウンドを再生
+*/
+// ==/WindhawkModSettings==
+
+#include <windows.h>
+#include <windowsx.h>
+#include <commctrl.h>
+
+// Settings structure
+struct ModSettings {
+    int triggerMode; // 0=mouse, 1=keyboard, 2=both
+    int mouseButton; // 0=left, 1=right, 2=middle
+    int clickType;   // 0=single, 1=double
+    bool useCtrl;
+    bool useAlt;
+    bool useShift;
+    int hotkey;      // Virtual key code
+    bool enableSound;
+} g_settings;
+
+// State variables
+static bool g_iconsHidden = false;
+static HWND g_desktopListView = nullptr;
+static DWORD g_lastClickTime = 0;
+static POINT g_lastClickPos = {0, 0};
+
+// Find desktop icons window
+HWND FindDesktopIconsWindow() {
```

Comment:
Have you considered that there can be several explorer.exe instances if the "launch folder windows in a separate process" option is enabled? In this case, non-desktop processes can find the windows of another process, which is probably not what you want.